feature/selector.py

```diff
@@ -270,6 +269,129 @@ def _validate(self):
                                                        CatBoostClassifier, CatBoostRegressor)),
                            ValueError("Unknown tree-based estimator" + str(self.estimator)))
 
+    class TextBased(NamedTuple):
+        """
+        # TODO  review the content to double check everything is %100 correct once ALL the implementation is done
```

Comment:
please revise

Reply:
let's get this into its final shape now..

Reply:
I re-check everything in PyDoc, just not sure about the following TODOs:
1) If no Y is given, it selects #TODO..
2) Randomness: Behavior for #TODO methods is non-deterministic, depends on seed.
for the second one, I explained what is the source of randomness in each optimization methods. I am not sure about the first one.